### DISCUSSION TITLE
clientside data fetching (with cookies)

### DISCUSSION THREAD

**Discussion by teremyx** [vercel/next.js#20789] [Help]
tl;dr: Client has to provide a cookie for authentication, but the cookies is needed for the api-backend and nextjs is used for the frontend. So when the nextjs app fetches data from the api, how can it include the user's cookie?

Hi. I use nextjs for the frontend and have a separate backend for the api (written in Kotlin using Ktor). 
The frontend is running under http://localhost:3000 and the backend is running on http://localhost:8080. 

The frontend makes (api-)requests using axios. 
For authentication I am using cookies, so when the frontend makes the axios.get-request it has to contain the cookies. 
I almost got it working: 

`export default function Test(props) {

 const comments = axios.get("http://localhost:8080/test", {
     withCredentials: true
    }).then((result) =>
      console.log(`comments: ${result.data}`
    ));
...
}
`

In the browser's developer console I can see the correct request being made (including the cookie), but a request is also made by the frontend nextjs server/app (which I can see in the node console). 
How can I prevent this additional api request by the nextjs server/app? This request also doesn't contain the cookie, since the cookie is stored in the browser and not in the nextjs app. 

I don't need Server-side Rendering (getServerSideProps) but how would I even do that, when the user/browser needs to provide a cookie, but for the api backend? Does the user also need to set a cookie for http://localhost:3000 (the nextjs app) and the nextjs app will then include it in the api call?

**Comment by teremyx** [comment]
If you scroll down to "Fetching data on client side" on this page: https://nextjs.org/docs/basic-features/data-fetching
It doesn't say _how_ to actually fetch data on client side. It says "Then, fetch the data on the client side and display it when ready."
How does that explain anything? I am still not sure how to do it.
